### PR TITLE
nitcatalog: handle local images

### LINKED ISSUE #2042: nitcatalog: no images in markdown
What to do with linked images?

e.g. http://nitlanguage.org/catalog/p/tinks.html

1. copy-them in the catalog?
2. hotlink-them from the gitweb/github ?

### PR DESCRIPTION
**Pull request by privat**
nitcatalog is extended to detect links to local images, copy the resource in the catalog directory and hijack the generated link.
This should fix #2042 and fix the images in http://nitlanguage.org/catalog/p/tinks.html

Most is done by monkey-patching (aka dirty refinement) since the doc_down model uses an internal private global markdown decorator. I'm not sure what should be the best approach here, and I had to improvise since @Morriar is MIA.

Bonus:

* `file_copy_to` is fixed and do not corrupt binary files (thanks @R4PaSs)
* `tests/test_prog` is used to test the catalog

### CODE REVIEW

File: src/nitcatalog.nit [hunk 1]
```diff
@@ -131,7 +131,68 @@ g.defer=true; g.async=true; g.src=u+'piwik.js'; s.parentNode.insertBefore(g,s);
...
+			fulllink.file_copy_to(res)
+
+			# Hijack the link in the html.
+			link = ".." / "res" / hash
+			super(v, link, name, comment)
+			return
+		end
+
+		# Something was bad
+		catalog.modelbuilder.toolcontext.error(current_mdoc.location, "Cannot found local image `{link}`")
```
Comment (xymus) [inline src/nitcatalog.nit]: Something *went bad

Cannot *find local image

**Comment by xymus** [comment]
I'm not very familiar with the code base but I'll use this feature for sure! +1

File: src/nitcatalog.nit [hunk 2]
```diff
@@ -131,7 +131,68 @@ g.defer=true; g.async=true; g.src=u+'piwik.js'; s.parentNode.insertBefore(g,s);
...
+			if mdoc == null then break
+			var source = mdoc.location.file
+			if source == null then break
+			var path = source.filename
+			var stat = path.file_stat
+			if stat == null then break
+			if not stat.is_dir then path = path.dirname
+
+			# Get the full path to the local resource
+			var fulllink = path / link.to_s
```
Comment (BlackMinou) [inline src/nitcatalog.nit]: plz can you name this var `full_link` ? It hurts my eyes...

**Comment by BlackMinou** [comment]
+1

**Pull request merged**